libphal/phal_dump.C

```diff
@@ -0,0 +1,418 @@
+			    dumpPath / dumpFilename;
+			try {
+				writeDumpFile(
+				    reinterpret_cast<char*>(&dumpRegs[0]),
+				    sizeof(DumpPIBMSRegVal) * dumpRegs.size(),
+				    basePath);
+			} catch (const dumpError_t& e) {
+				log(level::ERROR,
+				    "Error in writing p10_pibms_reg_dump file, "
+				    "errorMsg=%s",
+				    e.what());
+				throw;
+			}
+		}
+
+		// Dump the PIBMEM Array based on starting and number of address
+		std::vector<array_data_t> pibmemContents;
+		bool eccEnable = false;
+		uint32_t pibmemDumpStartByte = 0;
+		// Number of bytes to be read from PIBMEM
+		static constexpr uint32_t pibmemDumpNumOfByte = 0x7D400;
+		user_options userOptions = INTERMEDIATE_TILL_INTERMEDIATE;
+
+		fapiRc = p10_pibmem_dump(proc, pibmemDumpStartByte,
+					 pibmemDumpNumOfByte, userOptions,
+					 pibmemContents, eccEnable);
+		if (fapiRc != fapi2::FAPI2_RC_SUCCESS) {
+			log(level::ERROR,
+			    "Failed in p10_pibmem_dump for proc=%s, rc=0x%08X",
+			    pdbg_target_path(proc), fapiRc);
+		} else {
+			std::vector<uint64_t> dumpData;
+			for (auto& data : pibmemContents) {
+				dumpData.push_back(data.read_data);
+			}
+			std::string dumpFilename =
+			    baseFilename + "p10_pibmem_dump";
+			std::filesystem::path basePath =
+			    dumpPath / dumpFilename;
+			try {
+				writeDumpFile(
+				    reinterpret_cast<char*>(&dumpData[0]),
+				    sizeof(uint64_t) * dumpData.size(),
+				    basePath);
+			} catch (const dumpError_t& e) {
+				log(level::ERROR,
+				    "Error in writing p10_pibmem_dump file "
+				    "errorMsg=%s",
+				    e.what());
+				throw;
+			}
+		}
+
+		// Dump the PPE state based on the based base address
+		PPE_DUMP_MODE mode = SNAPSHOT;
+		uint32_t instanceNum = 0;
+		std::vector<Reg32Value_t> ppeGprsValue;
+		std::vector<Reg32Value_t> ppeSprsValue;
+		std::vector<Reg32Value_t> ppeXirsValue;
+		PPE_TYPES type = PPE_TYPE_SBE;
+
+		fapiRc =
+		    p10_ppe_state(proc, type, instanceNum, mode, ppeGprsValue,
+				  ppeSprsValue, ppeXirsValue);
+		if (fapiRc != fapi2::FAPI2_RC_SUCCESS) {
+			log(level::ERROR,
+			    "Failed in p10_ppe_state for proc=%s, rc=0x%08X",
+			    pdbg_target_path(proc), fapiRc);
+		} else {
+			std::vector<DumpPPERegValue> ppeState;
+			for (auto& gpr : ppeGprsValue) {
+				ppeState.emplace_back(gpr.number, gpr.value);
+			}
+			for (auto& spr : ppeSprsValue) {
+				ppeState.emplace_back(spr.number, spr.value);
+			}
+			for (auto& xir : ppeXirsValue) {
+				ppeState.emplace_back(xir.number, xir.value);
+			}
+
+			std::string dumpFilename =
+			    baseFilename + "p10_ppe_state";
+			std::filesystem::path basePath =
+			    dumpPath / dumpFilename;
+			try {
+				writeDumpFile(
+				    reinterpret_cast<char*>(&ppeState[0]),
+				    sizeof(DumpPPERegValue) * ppeState.size(),
+				    basePath);
+			} catch (const dumpError_t& e) {
+				log(level::ERROR,
+				    "Error in writing p10_ppe_state file, "
+				    "errorMsg=%s",
+				    e.what());
+				throw;
+			}
```

Comment:
log statements in writeDumpFile is sufficient I guess, just do not try..catch it , let it flow to external block.

Reply:
This will give better logging and the external block will do cleanup